smps.proto

```diff
@@ -15,34 +14,45 @@ message PerformanceTestConfig {
             tcp = 1;
             grpc = 2;
         }
-        
+
+        enum LoadGenerator {
+            fortio = 0;
+            wrk2 = 1;
```

Comment:
Is there a use case for tracking the load generator’s version number? I suppose there is, but perhaps, is not an essential use case.

Reply:
Hmm, I think we can have a string rather than enum here. @kanishkarj opinion ?

Reply:
We could, encoding the version number is easy if it is a string.

Reply:
@kushthedude were you exploring the version number as a string or the name of the load generator as a string?

Reply:
name of load_generator as a string, we don't have a field for version number (we shouldn't store it, should we ? )

Reply:
While technically, the version of the load generator can have a material affect (e.g. how it performs its results analysis), I think this is an edge use case and that the tracking of the version number of the load generator may be overkill.

Reply:
Just a query, so are we planning to support multiple-versions of load_generators?

Reply:
Having the load generator name as an enum means that we'll have to release a new version of the spec whenever a new load generator would like to be supported within SMPS, which is not a big deal. The value of having the load generator as an enum is that different implementations of SMPS can be guaranteed as compatible. If we hold implementation interoperability as a goal, we should lean into the use of an enum. 

**Example:** 
- User exports test results from Meshery to a file. Meshery uses the string "NightHawk" to refer to the Nighthawk load generator. 
- User imports those same results into another system that has implemented SMPS.  That system uses the string "Night-hawk" to refer to the Nighthawk load generator. Depending upon that system's design, the import fails or the result set fails to show as a "Nighthawk" test.

Reply:
Got it, then we should go ahead & also use enum for some variety of choices like as of now service-mesh(type).

Reply:
I agree. I made this same note right about the same time you mentioned it. I think a separate proto file for the long list of service mesh name enums may make sense.